From 6e5a4bf16f1bf0f2736451aef4ba34a5351040bb Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Tue, 19 Jul 2022 14:49:51 +0800
Subject: [PATCH 29/62] Revert "drm: msm: fix possible memory leak in
 mdp5_crtc_cursor_set()"

This reverts commit 656aa3c51fc662064f17179b38ec3ce43af53bca.
---
 drivers/gpu/drm/msm/disp/mdp5/mdp5_crtc.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/msm/disp/mdp5/mdp5_crtc.c b/drivers/gpu/drm/msm/disp/mdp5/mdp5_crtc.c
index ff4f207cbdea..06f19ef5dbf3 100644
--- a/drivers/gpu/drm/msm/disp/mdp5/mdp5_crtc.c
+++ b/drivers/gpu/drm/msm/disp/mdp5/mdp5_crtc.c
@@ -983,10 +983,8 @@ static int mdp5_crtc_cursor_set(struct drm_crtc *crtc,
 
 	ret = msm_gem_get_and_pin_iova(cursor_bo, kms->aspace,
 			&mdp5_crtc->cursor.iova);
-	if (ret) {
-		drm_gem_object_put(cursor_bo);
+	if (ret)
 		return -EINVAL;
-	}
 
 	pm_runtime_get_sync(&pdev->dev);
 
-- 
2.27.0

